app/registries/foreman/setting_manager.rb

```diff
@@ -71,6 +79,42 @@ def setting(name, default:, description:, type:, full_name: nil, value: nil, col
           encrypted: encrypted,
           options: options,
         }
+        validates(name, validate) if validate
+        storage[name.to_s]
+      end
+
+      def validates(name, validations)
+        if validations.is_a?(Proc)
+          validates_with name, ValueLambdaValidator, proc: validations
+          return
+        elsif validations.is_a?(Regexp)
+          validations = { format: { with: validations } }
+        elsif validations.is_a?(Symbol)
+          validations = { validations => true }
+        end
```

Comment:
if we want this method to mimic the activerecord API, this part probably belongs outside it?

Reply:
well I'd like to allow `setting('name', validate: :email)` and thats only shorthand for `setting('name'); validates('name', :email)`. But I've moved it to enable this only in the inline validations.